router/batchrouter/asyncdestinationmanager/bing-ads/util.go

```diff
@@ -348,27 +391,29 @@ func ProcessPollStatusData(records [][]string) map[string]map[string]struct{} {
 		if typeIndex < len(record) && strings.Contains(rowname, "Error") {
 			if clientIDIndex >= 0 && clientIDIndex < len(record) {
 				// expecting the client ID is present as jobId<<>>clientId
-				clientID := strings.Split(record[clientIDIndex], "<<>>")
-				if len(clientID) >= 2 {
-					errorSet, ok := clientIDErrors[clientID[0]]
-					if !ok {
-						errorSet = make(map[string]struct{})
-						// making the structure as jobId: [error1, error2]
-						clientIDErrors[clientID[0]] = errorSet
-					}
-					errorSet[record[errorIndex]] = struct{}{}
-
+				clientId, err := newClientIDFromString(record[clientIDIndex])
+				if err != nil {
+					return nil, fmt.Errorf("invalid client id: %s", err)
```

Comment:
return nil, err
(err already contains the right message)

Reply:
done